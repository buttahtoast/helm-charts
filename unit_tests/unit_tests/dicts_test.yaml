suite: library/dictionaries unit tests
tests:

  ## Dictionary (Get)
  
  - it: "[Dicts | Get]: Functionality (Primitive Type)"
    templates:
    - dicts/get.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.minReadySeconds"
    asserts:
      - matchSnapshot: {}

  - it: "[Dicts | Get]: Functionality (Map Type)"
    templates:
    - dicts/get.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.selector"
    asserts:
      - matchSnapshot: {}

  - it: "[Dicts | Get]: Functionality (Slice Type)"
    templates:
    - dicts/get.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.template.spec.containers"
    asserts:
      - matchSnapshot: {}

  - it: "[Dicts | Get]: Non Existent Path"
    templates:
    - dicts/get.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.nonExistent"
    asserts:
      - matchSnapshot: {}

  ## Dictionary (Set)

  - it: "[Dicts | Set]: Functionality (Primitive Type)"
    templates:
    - dicts/set.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.minReadySeconds"
      value: 10000
    asserts:
      - matchSnapshot: {}

  - it: "[Dicts | Set]: Functionality Recursive (Primitive Type)"
    templates:
    - dicts/set.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.subelement.new-element"
      value: "new-value"
    asserts:
      - matchSnapshot: {}

  - it: "[Dicts | Set]: Functionality Recursive (Slice Type)"
    templates:
    - dicts/set.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.subelement.new-element"
      value: 
        - element_1
        - element_2
    asserts:
      - matchSnapshot: {}

  - it: "[Dicts | Set]: Functionality Recursive (Map Type)"
    templates:
    - dicts/set.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.subelement.newElement"
      value: 
        some:
          map: "data"
        new:
          test: "value"
    asserts:
      - matchSnapshot: {}



  - it: "[Dicts | Set]: Functionality Recursive (Empty Data)"
    templates:
    - dicts/set.yaml
    values:
      - ./values/dicts/base.yaml
    set: 
      path: "spec.subelement.new-element"
      value: {}
    asserts:
      - matchSnapshot: {}      


  ## Check Dictionary Functions
#
#  - it: "[Dicts | Merge]: List Inject"
#    set:
#      Base:
#        spec:
#          selector:
#            app: nginx
#          ports:
#          - port: 80
#            name: http
#            targetPort: 80
#          - port: 443
#            name: https
#            targetPort: 80
#      Data:
#        spec:
#          ports:
#          - __inject__
#          - port: 8443
#            name: https
#          - port: 10000
#            name: metrics
#            targetPort: 9001
#    asserts:
#      - matchSnapshot: {}
#
#  - it: "[Dicts | Merge]: Merge Key"
#    set:
#      unit_test: 
#        chart: "library"
#        type: "dict.merge"
#      Base:
#        spec:
#          selector:
#            app: nginx
#          ports:
#          - port: 80
#            name: http
#            targetPort: 80
#          - port: 8443
#            name: https
#            targetPort: 80
#          - port: 9999
#            name: base_port
#            targetPort: 9999
#      Data:
#        spec:
#          ports:
#          - __inject__
#          - ((port))
#          - port: 8443
#            name: https_metrics
#            targetPort: 8443
#          - port: 80
#            name: http_port
#            targetPort: 80
#          - port: 8888
#            name: custom_port
#            targetPort: 8888
#    asserts:
#      - matchSnapshot: {}
#
#