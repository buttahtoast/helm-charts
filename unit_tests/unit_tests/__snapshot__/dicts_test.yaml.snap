'[Dicts | Get]: Functionality (Map Type)':
  1: |
    res:
      matchLabels:
        app: backend
'[Dicts | Get]: Functionality (Primitive Type)':
  1: |
    res: 3
'[Dicts | Get]: Functionality (Slice Type)':
  1: |
    res:
    - command:
      - ./podinfo
      - --port=9898
      - --port-metrics=9797
      - --grpc-port=9999
      - --grpc-service-name=backend
      - --level=info
      - --cache-server=cache:6379
      env:
      - name: PODINFO_UI_COLOR
        value: '#34577c'
      image: ghcr.io/stefanprodan/podinfo:6.1.1
      imagePullPolicy: IfNotPresent
      livenessProbe:
        exec:
          command:
          - podcli
          - check
          - http
          - localhost:9898/healthz
        initialDelaySeconds: 5
        timeoutSeconds: 5
      name: backend
      ports:
      - containerPort: 9898
        name: http
        protocol: TCP
      - containerPort: 9797
        name: http-metrics
        protocol: TCP
      - containerPort: 9999
        name: grpc
        protocol: TCP
      readinessProbe:
        exec:
          command:
          - podcli
          - check
          - http
          - localhost:9898/readyz
        initialDelaySeconds: 5
        timeoutSeconds: 5
      resources:
        limits:
          cpu: 2000m
          memory: 512Mi
        requests:
          cpu: 100m
          memory: 32Mi
'[Dicts | Get]: Non Existent Path':
  1: |
    res: {}
'[Dicts | Set]: Functionality (Primitive Type)':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: backend
    spec:
      minReadySeconds: 10000
      progressDeadlineSeconds: 60
      revisionHistoryLimit: 5
      selector:
        matchLabels:
          app: backend
      strategy:
        rollingUpdate:
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            prometheus.io/port: "9797"
            prometheus.io/scrape: "true"
          labels:
            app: backend
        spec:
          containers:
          - command:
            - ./podinfo
            - --port=9898
            - --port-metrics=9797
            - --grpc-port=9999
            - --grpc-service-name=backend
            - --level=info
            - --cache-server=cache:6379
            env:
            - name: PODINFO_UI_COLOR
              value: '#34577c'
            image: ghcr.io/stefanprodan/podinfo:6.1.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/healthz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            name: backend
            ports:
            - containerPort: 9898
              name: http
              protocol: TCP
            - containerPort: 9797
              name: http-metrics
              protocol: TCP
            - containerPort: 9999
              name: grpc
              protocol: TCP
            readinessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/readyz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            resources:
              limits:
                cpu: 2000m
                memory: 512Mi
              requests:
                cpu: 100m
                memory: 32Mi
'[Dicts | Set]: Functionality Recursive (Empty Data)':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: backend
    spec:
      minReadySeconds: 3
      progressDeadlineSeconds: 60
      revisionHistoryLimit: 5
      selector:
        matchLabels:
          app: backend
      strategy:
        rollingUpdate:
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            prometheus.io/port: "9797"
            prometheus.io/scrape: "true"
          labels:
            app: backend
        spec:
          containers:
          - command:
            - ./podinfo
            - --port=9898
            - --port-metrics=9797
            - --grpc-port=9999
            - --grpc-service-name=backend
            - --level=info
            - --cache-server=cache:6379
            env:
            - name: PODINFO_UI_COLOR
              value: '#34577c'
            image: ghcr.io/stefanprodan/podinfo:6.1.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/healthz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            name: backend
            ports:
            - containerPort: 9898
              name: http
              protocol: TCP
            - containerPort: 9797
              name: http-metrics
              protocol: TCP
            - containerPort: 9999
              name: grpc
              protocol: TCP
            readinessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/readyz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            resources:
              limits:
                cpu: 2000m
                memory: 512Mi
              requests:
                cpu: 100m
                memory: 32Mi
'[Dicts | Set]: Functionality Recursive (Map Type)':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: backend
    spec:
      minReadySeconds: 3
      progressDeadlineSeconds: 60
      revisionHistoryLimit: 5
      selector:
        matchLabels:
          app: backend
      strategy:
        rollingUpdate:
          maxUnavailable: 0
        type: RollingUpdate
      subelement: {}
      template:
        metadata:
          annotations:
            prometheus.io/port: "9797"
            prometheus.io/scrape: "true"
          labels:
            app: backend
        spec:
          containers:
          - command:
            - ./podinfo
            - --port=9898
            - --port-metrics=9797
            - --grpc-port=9999
            - --grpc-service-name=backend
            - --level=info
            - --cache-server=cache:6379
            env:
            - name: PODINFO_UI_COLOR
              value: '#34577c'
            image: ghcr.io/stefanprodan/podinfo:6.1.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/healthz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            name: backend
            ports:
            - containerPort: 9898
              name: http
              protocol: TCP
            - containerPort: 9797
              name: http-metrics
              protocol: TCP
            - containerPort: 9999
              name: grpc
              protocol: TCP
            readinessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/readyz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            resources:
              limits:
                cpu: 2000m
                memory: 512Mi
              requests:
                cpu: 100m
                memory: 32Mi
'[Dicts | Set]: Functionality Recursive (Primitive Type)':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: backend
    spec:
      minReadySeconds: 3
      progressDeadlineSeconds: 60
      revisionHistoryLimit: 5
      selector:
        matchLabels:
          app: backend
      strategy:
        rollingUpdate:
          maxUnavailable: 0
        type: RollingUpdate
      subelement:
        new-element: new-value
      template:
        metadata:
          annotations:
            prometheus.io/port: "9797"
            prometheus.io/scrape: "true"
          labels:
            app: backend
        spec:
          containers:
          - command:
            - ./podinfo
            - --port=9898
            - --port-metrics=9797
            - --grpc-port=9999
            - --grpc-service-name=backend
            - --level=info
            - --cache-server=cache:6379
            env:
            - name: PODINFO_UI_COLOR
              value: '#34577c'
            image: ghcr.io/stefanprodan/podinfo:6.1.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/healthz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            name: backend
            ports:
            - containerPort: 9898
              name: http
              protocol: TCP
            - containerPort: 9797
              name: http-metrics
              protocol: TCP
            - containerPort: 9999
              name: grpc
              protocol: TCP
            readinessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/readyz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            resources:
              limits:
                cpu: 2000m
                memory: 512Mi
              requests:
                cpu: 100m
                memory: 32Mi
'[Dicts | Set]: Functionality Recursive (Slice Type)':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: backend
    spec:
      minReadySeconds: 3
      progressDeadlineSeconds: 60
      revisionHistoryLimit: 5
      selector:
        matchLabels:
          app: backend
      strategy:
        rollingUpdate:
          maxUnavailable: 0
        type: RollingUpdate
      subelement:
        new-element:
        - element_1
        - element_2
      template:
        metadata:
          annotations:
            prometheus.io/port: "9797"
            prometheus.io/scrape: "true"
          labels:
            app: backend
        spec:
          containers:
          - command:
            - ./podinfo
            - --port=9898
            - --port-metrics=9797
            - --grpc-port=9999
            - --grpc-service-name=backend
            - --level=info
            - --cache-server=cache:6379
            env:
            - name: PODINFO_UI_COLOR
              value: '#34577c'
            image: ghcr.io/stefanprodan/podinfo:6.1.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/healthz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            name: backend
            ports:
            - containerPort: 9898
              name: http
              protocol: TCP
            - containerPort: 9797
              name: http-metrics
              protocol: TCP
            - containerPort: 9999
              name: grpc
              protocol: TCP
            readinessProbe:
              exec:
                command:
                - podcli
                - check
                - http
                - localhost:9898/readyz
              initialDelaySeconds: 5
              timeoutSeconds: 5
            resources:
              limits:
                cpu: 2000m
                memory: 512Mi
              requests:
                cpu: 100m
                memory: 32Mi
